src/analysis/calculus/mean_value.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2019 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+
+The mean value inequality: a bound on the derivative of a function implies that this function
+is Lipschitz continuous for the same bound.
+-/
+
+
+import analysis.calculus.deriv
+
+set_option class.instance_max_depth 90
```

Comment:
I always worry when I see something like this. Was elaboration really slow when you were working on the proofs?

Reply:
With a lower max_depth, Lean does not find some instances, so the proof does not go through. Elaboration was not particularly slow, even with the increased max_depth. In general, I don't mind increasing the max_depth, since it does not really hurt (unless there are problematic instances, but then they should be corrected anyway).